hologram/__init__.py

```diff
@@ -526,9 +602,44 @@ def _get_field_meta(cls, field: Field) -> Tuple[FieldMeta, bool]:
 
         return field_meta, required
 
+    @classmethod
+    def _encode_restrictions(
+        cls, restrictions: Union[List[Any], Enum]
+    ) -> JsonDict:
+        field_schema: JsonDict = {}
+        member_types = set()
+        values = []
+        for member in restrictions:
+            if isinstance(member, Enum):
+                value = member.value
+            else:
+                value = member
+            member_types.add(type(value))
+            values.append(value)
+        if len(member_types) == 1:
+            member_type = member_types.pop()
+            if member_type in JSON_ENCODABLE_TYPES:
+                field_schema.update(JSON_ENCODABLE_TYPES[member_type])
+            else:
+                field_schema.update(
+                    cls._field_encoders[member_type].json_schema
```

Comment:
This used to say `member_types.pop()` instead of `member_type`, but that would be popping from an empty set and I'm pretty confident this is what was intended.

I'm not sure anyone really should be making enums of types that aren't in `JSON_ENCODABLE_TYPES` in the first place, but I didn't see any reason to remove that support.

Reply:
Also, irritating thing: This check doesn't handle the case where multiple different StrEnums are provided in `restrict` that don't match the underlying field's type. I spent some time on that but I couldn't find a nice way to do it. So just don't do that, I guess :( Hopefully `mypy` will catch stuff like that.